code/game/gamemodes/dynamic/ruleset_picking.dm

```diff
@@ -83,6 +89,13 @@
 /datum/game_mode/dynamic/proc/execute_midround_latejoin_rule(sent_rule)
 	var/datum/dynamic_ruleset/rule = sent_rule
 	spend_midround_budget(rule.cost, threat_log, "[worldtime2text()]: [rule.ruletype] [rule.name]")
+	if (simulated)
+		if(rule.flags & ONLY_RULESET)
+			only_ruleset_executed = TRUE
+		executed_rules += rule
+		if (rule.persistent)
+			current_rules += rule
```

Comment:
what is this all about?

Reply:
If we are simulating the gamemode, then we want to set everything associated to spawning a gamemode without actually executing it. This will make the rule as being executed and do everything that normal spawning does but won't trigger pre execute or eecute